cli-plugins/apps/create-workload.md

```diff
@@ -15,81 +15,120 @@ The following prerequisites are required to use workloads with Tanzu Application
 
-Here is how you can get started with an example workload.
+### <a id="workload-git"></a> Create a workload from GitHub repository
 
-- To name the workload and specify a source code location to create the workload from, run:
+Tanzu Application Platform supports creating a workload from an existing git repository by setting the flags `--git-repo`, `--git-branch`, `--git-tag` and `--git-commit`, this will allow the out of the box [supply chain](../../scc/about.md) to get the source from the given repository to deploy the application.
 
+To create a named workload and specify a git source code location, run:
+
+ ```console
+tanzu apps workload create pet-clinic --git-repo https://github.com/sample-accelerators/spring-petclinic --git-tag tap-1.1 --type web  
+```
+
+Respond `Y` to prompts to complete process.
+
+Where:
+
+- `pet-clinic` is the name of the workload.
+- `--git-repo` is the location of the code to build the workload from.
+- `--git-branch` (optional) specifies which branch in the repository to pull the code from.
+- `--type` is used to distinguish the workload type.
+
+You can find the options available for specifying the workload in the command reference for [`workload create`](command-reference/tanzu-apps-workload-create.md), or you can run `tanzu apps workload create --help`.
+
+### <a id="workload-local-source"></a> Create a workload from local source code
+
+Tanzu Application Platform supports creating a workload from an existing local project by setting the flags `--local-path` and `--source-image`, then this will allow the out of the box [supply chain](../../scc/about.md) to generate an image ([carvel-imgpkg](https://carvel.dev/imgpkg/)) and push it to the given registry to be used in the workload.
+
+- To create a named workload and specify where the local source code is, run:
 
     ```console
-    tanzu apps workload create pet-clinic --git-repo https://github.com/sample-accelerators/spring-petclinic --git-tag tap-1.1 --type web  
+    tanzu apps workload create pet-clinic --local-path /path/to/my/project --source-image springio/petclinic
     ```
 
-    Respond `Y` to prompts to complete process.
+    Respond `Y` to the prompt about publishing local source code if the image needs to be updated.
 
     Where:
 
     - `pet-clinic` is the name of the workload.
-    - `--git-repo` is the location of the code to build the workload from.
-    - `--git-branch` (optional) specifies which branch in the repository to pull the code from.
-    - `--type` is used to distinguish the workload type.
-
-    You can find the options available for specifying the workload in the command reference for [`workload create`](command-reference/tanzu-apps-workload-create.md), or you can run `tanzu apps workload create --help`.
+    - `--local-path` points to the directory where the source code is located.
+    - `--source-image` is the registry path when the local source code will be uploaded as an image.
```

Comment:
should be 
```
- `--source-image` is the registry path where the local source code will be uploaded as an image.
```

use "where" instead of "when"